designs/2022-lintFiles-lintText-fix-support/README.md

```diff
@@ -0,0 +1,251 @@
+        ...unknownOptions
+    } = options || {};
+
+    // ...
+
+    const { cliEngine } = privateMembersMap.get(this);
+
+    return processCLIEngineLintReport(
+        cliEngine,
+        cliEngine.executeOnText(code, filePath, warnIgnored, fix)
+    );
+}
+```
+
+`cliEngine.executeOnText` merges the fix passed in `lintText()` and the `options.fix` in `ESLint.constructor()` (I omit irrelevant code):
+
+```javascript
+executeOnText(text, filename, warnIgnored, fixFromLintText) {
+    const {
+        options: {
+            fix: fixFromESLint,
+        }
+    } = internalSlotsMap.get(this);
+
+        // Do lint.
+        results.push(verifyText({
+            // text,
+            // filePath: resolvedFilename,
+            // config,
+            // cwd,
+            fix: fixFromLintText !== undefined ? fixFromLintText : fixFromESLint,
+            // allowInlineConfig,
+            // reportUnusedDisableDirectives,
+            // fileEnumerator,
+            // linter
+        }));
+}
+```
+
+### eslint.lintFiles(patterns)
+
+`eslint.lintFiles(patterns)` adds options parameter and provides fix parameter:
+
+```javascript
+async lintFiles(patterns, options = {}) {
+    // ...
+
+    const {
+        fix,
+        ...unknownOptions
+    } = options || {};
+
+    const unknownOptionKeys = Object.keys(unknownOptions);
+
+    if (unknownOptionKeys.length > 0) {
+        throw new Error(`'options' must not include the unknown option(s): ${unknownOptionKeys.join(", ")}`);
+    }
+
+    const { cliEngine } = privateMembersMap.get(this);
+
+    return processCLIEngineLintReport(
+        cliEngine,
+        cliEngine.executeOnFiles(patterns, fix)
+    );
+}
+```
+
+`cliEngine.executeOnFiles` has basically the same changes as `cliEngine.executeOnText` above:
+
+```javascript
+executeOnFiles(text, fixFromLintFiles) {
+    const {
+        options: {
+            fix: fixFromESLint,
+        }
+    } = internalSlotsMap.get(this);
+
+        // Do lint.
+        results.push(verifyText({
+            // text,
+            // filePath: resolvedFilename,
+            // config,
+            // cwd,
+            fix: fixFromLintFiles !== undefined ? fixFromLintFiles : fixFromESLint,
+            // allowInlineConfig,
+            // reportUnusedDisableDirectives,
+            // fileEnumerator,
+            // linter
+        }));
+}
+```
+
+## Documentation
+
+No.
+
+## Drawbacks
+
+<!--
+    Why should we *not* do this? Consider why adding this into ESLint
+    might not benefit the project or the community. Attempt to think 
+    about any opposing viewpoints that reviewers might bring up. 
+
+    Any change has potential downsides, including increased maintenance
+    burden, incompatibility with other tools, breaking existing user
+    experience, etc. Try to identify as many potential problems with
+    implementing this RFC as possible.
+-->
+
+I need both fix and no-fix in one process, I currently have to initialize two eslint instances:
```

Comment:
This section should describe why we WOULD NOT want to implement this feature. Here you have just restated why you WOULD want to implement this. Please think a bit about the drawbacks of implementing this RFC.

Reply:
Drawbacks? What drawbacks. It's backward compatible and definitely it can never be removed from the constructor. So, no breaking things or bad things.

Reply:
This is part of evaluating any feature. Every change has a drawback even if it’s small. Maybe it can confuse users, if it’s a breaking change it will require people to change their code, if it’s not, there are other points of confusion. Please think more about this.